cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/jmx/JmxConnect.java

```diff
@@ -0,0 +1,251 @@
+    private final int port;
+    private final String username;
+    private final String password;
+
+    private JMXConnector jmxc;
+    private MBeanServerConnection mbeanServerConn;
+
+    private StorageServiceMBean ssProxy;
+    private MessagingServiceMBean msProxy;
+    private StreamManagerMBean streamProxy;
+    private CompactionManagerMBean compactionProxy;
+    private FailureDetectorMBean fdProxy;
+    private CacheServiceMBean cacheService;
+    private StorageProxyMBean spProxy;
+    private HintedHandOffManagerMBean hhProxy;
+    private GCInspectorMXBean gcProxy;
+    private GossiperMBean gossProxy;
+    private EndpointSnitchInfoMBean snitchProxy;
+    private MemoryMXBean memProxy;
+    private RuntimeMXBean runtimeProxy;
+
+    private final ReentrantLock lock = new ReentrantLock();
+
+    public JmxConnect(String host, int port, String username, String password) {
+        if (host == null || host.trim().isEmpty())
+            host = DEFATULT_CASSANDRA_JMX_HOST;
+        if (port <= 0)
+            port = DEFAULT_CASSANDRA_JMX_PORT;
+
+        this.host = host;
+        this.port = port;
+        this.username = username;
+        this.password = password;
+    }
+
+    public JmxConnect(String host, int port) {
+        this(host, port, null, null);
+    }
+
+    @Override
+    public void close() throws IOException {
+        lock.lock();
+        try {
+            if (jmxc == null)
+                return;
+
+            jmxc.close();
+        } finally {
+            lock.unlock();
+
+            jmxc = null;
+            mbeanServerConn = null;
+            ssProxy = null;
+            msProxy = null;
+            streamProxy = null;
+            compactionProxy = null;
+            fdProxy = null;
+            cacheService = null;
+            spProxy = null;
+            hhProxy = null;
+            gcProxy = null;
+            gossProxy = null;
+            snitchProxy = null;
+            memProxy = null;
+            runtimeProxy = null;
+        }
+    }
+
+    public void connect() throws IOException {
+        lock.lock();
+        try {
+            if (jmxc != null)
+                return;
+
+            JMXServiceURL jmxUrl = new JMXServiceURL(String.format(fmtUrl, host, port));
+            Map<String, Object> env = new HashMap<String, Object>();
+            if (username != null) {
+                String[] creds = {username, password};
+                env.put(JMXConnector.CREDENTIALS, creds);
+            }
+            jmxc = JMXConnectorFactory.connect(jmxUrl, env);
+            mbeanServerConn = jmxc.getMBeanServerConnection();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public <T> T newProxy(String name, Class<T> type) {
+        lock.lock();
+        try {
+            connect();
+            return JMX.newMBeanProxy(mbeanServerConn, new ObjectName(name), type);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Can we create a custom runtime exception so that we can more easily catch errors related to JMX Connections?